### PR TITLE
Bugfix/pwa 3377 after login issue on order success page

### PR DESCRIPTION
**Pull request by del15881**
## Description
This pr Adds the functionality to redirect a guest user to the account-information page when they register an account from the order-confirmation/order-success page. 

On login from the order-confirmation/order-success page guest users will be redirected to /order-history page.

## Related Issue
https://jira.corp.adobe.com/browse/PWA-3377

Closes #PWA-3377 (https://jira.corp.adobe.com/browse/PWA-3377)

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by glo82145 — APPROVED** [review]

**Comment by glo82145** [comment]
run scaffold-pwa

File: packages/peregrine/lib/talons/CheckoutPage/OrderConfirmationPage/useOrderConfirmationPage.js [hunk 1]
```diff
@@ -34,8 +36,26 @@ export const flattenCustomerOrderData = data => {
...
     }
+
     const { customer } = data;
-    const order = customer.orders.items[0];
+    const order = customer?.orders?.items?.[0];
+    if (!order || !order.shipping_address) {
+        // Return an empty response if no valid order or shipping address exists
+        return;
+
+        //Below code can be uncommented in case of wanting to show the order success page with blank shipping address.
```
Comment (glo82145) [inline packages/peregrine/lib/talons/CheckoutPage/OrderConfirmationPage/useOrderConfirmationPage.js:46]: remove these commented lines

**Comment by glo42707** [comment]
run code-build

**Review by glo42707 — APPROVED** [review]
approved

**Pull request merged**